cpp/src/Slice/Parser.cpp

```diff
@@ -1003,6 +998,31 @@ Slice::Contained::parseFormatMetaData(const list<string>& metaData)
     return result;
 }
 
+bool
+Slice::Contained::isDeprecated(bool checkParent) const
+{
+    const string deprecate = "deprecate";
```

Comment:
Sure, I'll do that in a follow-up PR though.
I want to avoid going too far down that rabbit hole though.
It'd probably take months to clean and modernize the compilers...